walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -29,21 +29,32 @@
     if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
       return ret;
     }
-    List<TimeRange> occupiedBlocks = recordOccupiedBlocks(events, request.getAttendees());
-    ret = complement(TimeRange.WHOLE_DAY, occupiedBlocks);
+    List<TimeRange> occupiedMandatory = new ArrayList<>();
+    List<TimeRange> occupiedAll = new ArrayList<>();
+    recordOccupiedBlocks(events, request.getAttendees(), request.getOptionalAttendees(),
+            occupiedMandatory, occupiedAll);
+    ret = complement(TimeRange.WHOLE_DAY, occupiedAll);
+    ret.removeIf(tr -> tr.duration() < request.getDuration());
```

Comment:
NIT: consider longer variable names, ret is ok but tr might be a tad confusing

Reply:
done!

Reply:
done!